src/sentry/api/endpoints/release_thresholds/health_checks/is_crash_free_rate_healthy.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import annotations
+
+import logging
+from datetime import datetime
+from typing import TYPE_CHECKING, Any, Dict, Tuple
+
+from dateutil import parser
+
+from sentry.models.release_threshold.constants import TriggerType
+
+from ..constants import CRASH_USERS_DISPLAY
+
+logger = logging.getLogger("sentry.release_threshold_status.is_crash_free_rate_healthy")
+
+if TYPE_CHECKING:
+    from sentry.api.endpoints.release_thresholds.types import EnrichedThreshold
+
+
+def get_groups_totals(
+    sessions_data: Dict[str, Any],
+    release_version: str,
+    project_id: int,
+    field: str,
+    start_idx: int,
+    end_idx: int,
+    environment: str | None = None,
+    status: str | None = None,
+) -> int:
+    filters = ["release", "project"]
+    if environment:
+        filters.append("environment")
+    if status:
+        filters.append("session.status")
+    group_series: list[dict[str, Any]] = [
+        g.get("series", {})
+        for g in sessions_data.get("groups", [])
+        if g.get("by", {}).get("release") == release_version
+        and g.get("by", {}).get("project") == project_id
+        and (g.get("by", {}).get("environment") == environment if environment else True)
+        and (g.get("by", {}).get("session.status") == status if status else True)
+    ]
+    total = 0
+    for group in group_series:
+        series = group.get(field, [])
+        if len(series) < start_idx or len(series) < end_idx:
+            logger.warning(
+                "Error calculating session totals",
+                extra={"start_idx": start_idx, "end_idx": end_idx, "series_length": len(series)},
+            )
+            raise IndexError("Start/End indexes are out of range for series")
+        total += sum(series[start_idx : end_idx + 1])
+    return total
+
+
+def get_interval_indexes(intervals: list[str], start: datetime, end: datetime) -> Tuple[int, int]:
+    """
+    :param intervals: if timestamps from fetched sessions data
+    :param start: timestamp
+    :param end: timestamp
+
+    :return: If any subsection of the intervals fall within the given start/end, return the indexes of both the start and the end
```

Comment:
for my understanding -- this gets the start and end indices of the `start` and `end` params within the list of intervals?

Reply:
Yes exactly.

In the sessions response, we get back a bunch of groups which each ahve their list of interval counts
but we also get back a list of intervals with their timestamps.

So - here we're detmerining which intervals from the interval groups should we focus on

Reply:
You can see an example of the repsonse shape in the `test_fixtures.py` file